src/BlockTypeRegistry.cpp

```diff
@@ -0,0 +1,229 @@
+
+#include "Globals.h"
+#include "BlockTypeRegistry.h"
+
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// BlockInfo:
+
+BlockInfo::BlockInfo(
+	const AString & aPluginName,
+	const AString & aBlockTypeName,
+	std::shared_ptr<cBlockHandler> aHandler,
+	const std::map<AString, AString> & aHints,
+	const std::map<AString, BlockInfo::HintCallback> & aHintCallbacks
```

Comment:
I'm conflicted whether to use `map` or `unordered_map` here. Considering the number of elements we expect this to have, I'd say it's an unnecessary detail right now.